### PR TITLE
refactor(.): cleanup redundancy

### PR DESCRIPTION
**Pull request by grandwizard28**
#### Refactors

- Add `.gitignore` file
- Correct dependencies of clickhouse on signoz
- Remove k8s-infra and all other resources from signoz
- Rename main container names to `deployment` name

### CODE REVIEW

File: charts/signoz/templates/otel-collector/deployment.yaml [hunk 1]
```diff
@@ -86,7 +86,7 @@ spec:
             {{- toYaml .Values.otelCollector.initContainers.init.resources | nindent 12 }}
         {{- end }}
       containers:
-        - name: {{ template "otelCollector.fullname" . }}
+        - name: app
```
Comment (prashant-shahi) [inline charts/signoz/templates/otel-collector/deployment.yaml:89]: Any specific reason to name otel-collector container `app`?

File: charts/signoz/Chart.yaml [hunk 2]
```diff
@@ -22,27 +22,7 @@ dependencies:
...
-    version: 4.3.0
-  - name: minio
-    repository: https://charts.min.io
-    condition: minio.enabled
-    version: 4.0.9
-  - name: keycloak
-    repository: https://charts.bitnami.com/bitnami
-    condition: keycloak.enabled
-    version: 10.1.5
+    version: 24.1.4
```
Comment (prashant-shahi) [inline charts/signoz/Chart.yaml:25]: Can you bump this to `24.1.5`?
I have a small change that can go along with this. - https://github.com/SigNoz/charts/pull/478

```suggestion
    version: 24.1.5
```

File: charts/signoz/Chart.lock [hunk 3]
```diff
@@ -1,21 +1,6 @@
...
-  version: v1.9.1
-- name: ingress-nginx
-  repository: https://kubernetes.github.io/ingress-nginx
-  version: 4.3.0
-- name: minio
-  repository: https://charts.min.io
-  version: 4.0.9
-- name: keycloak
-  repository: https://charts.bitnami.com/bitnami
-  version: 10.1.5
```
Comment (prashant-shahi) [inline charts/signoz/Chart.lock:19]: Nice job cleaning these up. I wanted to get rid of these for sometime now as well.

There will be dependent task of updating docs for keycloak SSO and secure signoz.

Also, with this there will be no infra metrics or logs collection by default with the SigNoz installation.
We will need to update docs accordingly.

Comment (grandwizard28) [reply charts/signoz/Chart.lock:19]: Can you please point me to the docs? I will update them!

File: charts/signoz/Chart.yaml (hunk 2 above)
Comment (grandwizard28) [reply charts/signoz/Chart.yaml:25]: Let's merge that change first!

File: charts/signoz/templates/otel-collector/deployment.yaml (hunk 1 above)
Comment (grandwizard28) [reply charts/signoz/templates/otel-collector/deployment.yaml:89]: I wanted to go with the name `app` (application) for all main containers!

File: charts/signoz/Chart.lock (hunk 3 above)
Comment (prashant-shahi) [reply charts/signoz/Chart.lock:19]: - Keycloak - https://signoz.io/docs/tutorial/setting-up-sso-saml-with-keycloak
- Secure SigNoz - https://signoz.io/docs/tutorial/setting-up-tls-for-signoz
- All kubernetes installation docs (EKS/GKE/Others) - https://signoz.io/docs/install/kubernetes

**Review by prashant-shahi — APPROVED** [review]

**Review by prashant-shahi — APPROVED** [review]

**Pull request merged**